src/weapon/w_011.c

```diff
@@ -155,8 +158,218 @@ int GetWeaponId(void) { return 11; }
+            g_api.CheckCollision(xVar, yVar, &collider, 0);
+            if ((collider.effects & EFFECT_SOLID) &&
+                !(collider.effects & EFFECT_UNK_8000)) {
+                goto make_other_entity;
+            }
+            xOffset = 8;
+            if (self->velocityX < 0) {
+                xOffset = -xOffset;
+            }
+            yOffset = 8;
+            xVar = self->posX.i.hi + xOffset;
+            yVar = self->posY.i.hi + yOffset;
+            g_api.CheckCollision(xVar, yVar, &collider, 0);
+            if ((collider.effects & EFFECT_SOLID) &&
+                !(collider.effects & EFFECT_UNK_8000)) {
+                goto make_other_entity;
+            }
+        }
+        xOffset = 8;
+        if (self->velocityX < 0) {
+            xOffset = -xOffset;
+        }
+        yOffset = -8;
+        xVar = self->posX.i.hi + xOffset;
+        yVar = self->posY.i.hi + yOffset;
+        g_api.CheckCollision(xVar, yVar, &collider, 0);
+        if ((collider.effects & EFFECT_SOLID) || (self->hitFlags)) {
+            goto make_other_entity;
+        }
+        if (g_Timer % 4 == 0) {
+            // Blueprint 90 is weapon entity 9, func_ptr_80170024, so ourself.
+            unusedEnt = g_api.CreateEntFactoryFromEntity(
+                self, ((g_HandId + 1) << 0xE) + FACTORY(0x100, 90), 0);
+        }
+        break;
+    case 2:
+        SetSpeedX(FIX(2.5));
+        if ((self->ext.timer.t % 10) == 0) {
+            g_api.PlaySfx(SFX_UNK_64B);
+        }
+        xOffset = 0;
+        yOffset = 25;
+        xVar = self->posX.i.hi + xOffset;
+        yVar = self->posY.i.hi + yOffset;
+        g_api.CheckCollision(xVar, yVar, &collider, 0);
+        if (collider.effects & EFFECT_SOLID) {
+            self->posY.i.hi += collider.unk18;
+        } else {
+            self->step = 1;
+            self->velocityY = 0;
+        }
+        if (!(collider.effects & EFFECT_UNK_8000)) {
+            xOffset = 8;
+            if (self->velocityX < 0) {
+                xOffset = -xOffset;
+            }
+            yOffset = 22;
+            xVar = self->posX.i.hi + xOffset;
+            yVar = self->posY.i.hi + yOffset;
+            g_api.CheckCollision(xVar, yVar, &collider, 0);
+            if ((collider.effects & EFFECT_SOLID) &&
+                !(collider.effects & EFFECT_UNK_8000)) {
+                goto make_other_entity;
+            }
+            xOffset = 8;
+            if (self->velocityX < 0) {
+                xOffset = -xOffset;
+            }
+            yOffset = 8;
+            xVar = self->posX.i.hi + xOffset;
+            yVar = self->posY.i.hi + yOffset;
+            g_api.CheckCollision(xVar, yVar, &collider, 0);
+            if ((collider.effects & EFFECT_SOLID) &&
+                !(collider.effects & EFFECT_UNK_8000)) {
+                goto make_other_entity;
+            }
+        }
+        if ((((collider.effects & (EFFECT_UNK_8000 | EFFECT_UNK_4000)) ==
```

Comment:
```suggestion
        if ((((collider.effects & (EFFECT_UNK_8000 | EFFECT_UNK_4000)) ==
```
I wonder if extracting `(EFFECT_UNK_8000 | EFFECT_UNK_4000)` into a `const` or `#define` would make that if statement less of a nightmare to read

Reply:
Maybe I'll just make an UNK_C000 and then we can name it properly once we know what each of the individual 8000 and 4000 are